companion/src/helpers.cpp

```diff
@@ -369,35 +370,70 @@ void startSimulation(QWidget * parent, RadioData & radioData, int modelIdx)
     simuData->setCurrentModel(modelIdx);
   }
 
-  SimulatorMainWindow * dialog = new SimulatorMainWindow(parent, simulatorId, flags);
-  dialog->setWindowModality(Qt::ApplicationModal);
-  dialog->setAttribute(Qt::WA_DeleteOnClose);
+  // the directory will be automatically deleted when QTemporaryDir goes out of scope
+  QTemporaryDir tmpDir(QDir::tempPath() + "/etx-XXXXXX");
+  if (tmpDir.isValid()) {
+    qDebug() << "Created temporary settings directory" << tmpDir.path();
+  }
+  else {
+    QString resultMsg = QCoreApplication::translate("Companion", "Error creating temporary directory for models and settings.");
+    QMessageBox::critical(NULL, QCoreApplication::translate("Companion", "Simulator Error"), resultMsg);
+    return;
+  }
 
-  QObject::connect(dialog, &SimulatorMainWindow::destroyed, [simuData] (void) {
-#ifdef __APPLE__
-    simulatorRunning = false;
+  simuData->fixModelFilenames();
+  Storage storage(tmpDir.path());
+  if (!storage.write(*simuData)) {
+    QString resultMsg = QCoreApplication::translate("Companion", "Error writing models and settings to temporary directory.");
+    QMessageBox::critical(NULL, QCoreApplication::translate("Companion", "Simulator Error"), resultMsg);
+    return;
+  }
+
+  const QString path = QCoreApplication::applicationDirPath();
+
+  #if Q_OS_WIN
```

Comment:
Might need `#include <QtGlobal>` for Qt 5.15 https://doc.qt.io/qt-5/qtglobal.html and `#include <QtSystemDetection>`  according to https://doc.qt.io/qt-6/qtsystemdetection.html#Q_OS_WIN for Qt 6.8

Reply:
2 out of 3 compile just gotta love consistency

Reply:
Added include so let's see if success

Reply:
Can you try `#if defined (Q_OS_WIN)` instead?